core/arch/arm/kernel/generic_boot.c

```diff
@@ -506,11 +506,32 @@ static int add_dt_overlay_fragment(void *fdt, int offs)
 	return offs;
 }
 
+static int init_dt_overlay(void *fdt, int fdtsize)
+{
+	int fragment;
+	int ret;
+
+	/* A valid DTB has been passed as a parameter */
+	ret = fdt_check_header(fdt);
+	if (!ret) {
+		fdt_for_each_subnode(fragment, fdt, 0)
+			__dt_frag_id += 1;
+		return ret;
+	}
+
+	/* No DTB passed - create a new tree */
+	return fdt_create_empty_tree(fdt, fdtsize);
```

Comment:
An invalid DTB should rather trig an error (`panic()`) than be treated as if early boot did not provide a DTB.

Maybe call `fdt_create_empty_tree(fdt, fdtsize);` only upon `CFG_DT_ADDR` definition as stated in the commit comment.

Reply:
@etienne-lms what I want to support here is not having to pass an overlay from ATF at all.

Reply:
edit:
"Maybe call fdt_create_empty_tree(fdt, fdtsize); only upon CFG_DT_ADDR definition as stated in the commit comment."
Yes, OK that works for both objectives, I'll do that.

Reply:
I wonder if this is the right way.
Here a DTB with invalid data is silently ignored instead of raising some error.
And if the expect DTB area was not initialized,  `fdt_check_header()` may not successfully fail.

I hardly see how platform can guess if `CFG_DT_ADDR` points to a valid DTB, a existing DTB overlay or an uninitialized memory area.